src/Master.cpp

```diff
@@ -148,6 +148,17 @@ std::string Master::getCurrentFileName() const {
   return CurrentStatus.Filename;
 }
 
+std::string Master::getCurrentFilePath() const {
+  std::lock_guard LockGuard(StatusMutex);
+  std::string fullFilePath = MainConfig.getHDFOutputPrefix();
+  if (fullFilePath.empty()) {
+    fullFilePath = CurrentStatus.Filename;
+  } else {
+    fullFilePath = fullFilePath + "/" + CurrentStatus.Filename;
```

Comment:
For building the path in a more portable manner, consider using std::filesystem::path and the `/` operator: https://en.cppreference.com/w/cpp/filesystem/path/append